src/mode/liquid_highlight_rules.js

```diff
@@ -1,103 +1,267 @@
-            regex : "[+-]?\\d+(?:(?:\\.\\d*)?(?:[eE][+-]?\\d+)?)?\\b"
-        }, {
-            token : "constant.language.boolean",
-            regex : "(?:true|false)\\b"
-        }, {
-            token : keywordMapper,
-            regex : "[a-zA-Z_$][a-zA-Z0-9_$]*\\b"
-        }, {
-            token : "keyword.operator",
-            regex : "/|\\*|\\-|\\+|=|!=|\\?\\:"
-        }, {
-            token : "paren.lparen",
-            regex : /[\[\({]/
-        }, {
-            token : "paren.rparen",
-            regex : /[\])}]/
-        }, {
-            token : "text",
-            regex : "\\s+"
-        }]
-    });
-
-    this.normalizeRules();
-};
+  }
+
+  /* -------------------------------------------- */
+  /* EMBEDDED REGIONS                             */
+  /* -------------------------------------------- */
+
+  this.embedRules(JsonHighlightRules, "schema-", [
+    {
+      token: "schema-start",
+      next: "pop",
+      regex: /({%-?\s*\bendschema\b\s*-?%})/,
+      onMatch: onMatchEmbedded("endschema"),
+    },
+  ]);
+
+  this.embedRules(JavaScriptHighlightRules, "javascript-", [
+    {
+      token: "javascript-start",
+      next: "pop",
+      regex: /({%-?\s*\bendjavascript\b\s*-?%})/,
+      onMatch: onMatchEmbedded("endjavascript"),
+    },
+  ]);
+
+
+
+  this.embedRules(CssHighlightRules, "style-", [
+    {
+      token: "style-start",
+      next: "pop",
+      regex: /({%-?\s*\bendstyle\b\s*-?%})/,
+      onMatch: onMatchEmbedded("endstyle"),
+    },
+  ]);
+
+  this.embedRules(CssHighlightRules, "stylesheet-", [
+    {
+      token: "stylesheet-start",
+      next: "pop",
+      regex: /({%-?\s*\bendstylesheet\b\s*-?%})/,
+      onMatch: onMatchEmbedded("endstylesheet"),
+    },
+  ]);
+
+  /* -------------------------------------------- */
+  /* LIQUID GRAMMARS                              */
+  /* -------------------------------------------- */
+
+  this.addRules({
+    "liquid-start": [
+      {
+        token: "punctuation.close.output",
+        regex: /}}/,
+        next: "pop",
+      },
+      {
+        token: "punctuation.close.tag",
+        regex: /%}/,
+        next: "pop",
+      },
+      {
+        token: "string",
+        regex: /['](?:(?:\\.)|(?:[^'\\]))*?[']/,
+      },
+      {
+        token: "string",
+        regex: /["](?:(?:\\.)|(?:[^'\\]))*?["]/,
+      },
+      {
+        token: "constant.numeric",
+        regex: /0[xX][0-9a-fA-F]+\b/,
+      },
+      {
+        token: "constant.numeric",
+        regex: /[+-]?\d+(?:(?:\.\d*)?(?:[eE][+-]?\d+)?)?\b/,
+      },
+      {
+        token: "keyword.operator",
+        regex: /\||\*|\-|\+|=|!=|\?\|\:/,
+      },
+      {
+        token: "constant.language.boolean",
+        regex: /(?:true|false|nil|empty)\b/,
+      },
+      {
+        token: "keyword.operator",
+        regex: /(?:and|contains|in|with)\b/,
+      },
+      {
+        token: "keyword.block",
+        regex: /(?<={%-?)\s*([a-zA-Z_$][a-zA-Z0-9_$]+\b)/,
```

Comment:
The tests are failing because node 12 doesn't support backreference in regexp, could we remove backreferences by using an additional state, or matching both the brace and the word in one rule with array token?